packages/contracts/contracts/voting/majority/MajorityVotingBase.sol

```diff
@@ -237,10 +241,14 @@ abstract contract MajorityVotingBase is
     }
 
     /// @inheritdoc IMajorityVoting
-    function vote(uint256 _proposalId, VoteOption _voteOption, bool _tryEarlyExecution) public {
+    function vote(
+        uint256 _proposalId,
+        VoteOption _voteOption,
+        bool _tryEarlyExecution
+    ) public {
         address account = _msgSender();
 
-        if (_voteOption != VoteOption.None && !_canVote(_proposalId, account)) {
+        if (!_canVote(_proposalId, account)) {
```

Comment:
Isn't it simpler to simply `return` out of the function if the vote was None?

Reply:
The main question that we should answer first is if it should be 
**a)** allowed to change your vote back to `None` in `VotingMode
.VoteReplacement` or 
**b)** not allowed. 
 E.g., changing a vote from `Abstain` to `None` means that you withdrew your vote.   
 
 If we go for **a)** we don’t need to change the code.
   If we go for **b)** we should revert the `vote()` method call like this:
   
```solidity
   /// @inheritdoc IMajorityVoting
function vote(
    uint256 _proposalId,
    VoteOption _voteOption,
    bool _tryEarlyExecution
) public {
    address account = _msgSender();
    if (_voteOption == VoteOption.None) {
        revert VoteOptionNoneNotAllowed();
    }
    if (!_canVote(_proposalId, account)) {
        revert VoteCastForbidden({proposalId: _proposalId, account: account});
    }
    _vote(_proposalId, _voteOption, account, _tryEarlyExecution);
}
```   
      
     We should not let someone call the public `vote()` method that silently does not vote.

Reply:
As discussed, withdrawing a vote is not allowed, but we deal with this on APP-1479